Implement BASE_PATH environment variable

ItsSpaceMan opened the pull request:
PR to implement base path, such that the service can run on `/kutt`, `/url-shortener`, or any other base path specified. Ran through the application several times with BASE_PATH omitted and set, and everything is looking good.

Its-SpaceMan committed:
Implement BASE_PATH environment variable

poeti8 commented:
@ItsSpaceMan Yes

Its-SpaceMan committed:
Implement SHORT_URLS_INCLUDE_PATH environment variable; update readme and example env

ItsSpaceMan committed:
Merge branch 'thedevs-network:main' into basepath

Its-SpaceMan committed:
Reenable main page due to functional changes for base paths

yq2028 reviewed (commented):
in file : server/utils/utils.js , this snippet is wrong.  it make the short link include the base_path no matter what the env.SHORT_URLS_INCLUDE_PATH is true or false.
    if (linkDomain === env.DEFAULT_DOMAIN) {
      path = env.BASE_PATH;

Its-SpaceMan committed:
Merge remote-tracking branch 'upstream/main' into basepath

Its-SpaceMan committed:
Streamline getShortURL logic

ItsSpaceMan commented:
> in file : server/utils/utils.js , this snippet is wrong. it make the short link include the base_path no matter what the env.SHORT_URLS_INCLUDE_PATH is true or false. if (linkDomain === env.DEFAULT_DOMAIN) { path = env.BASE_PATH;

Good catch; cleaned up. Thanks @yq2028.

Added a new commit as well to fix up some main page logic (see [3710b71](https://github.com/thedevs-network/kutt/pull/890/commits/3710b71c80332ead134bd7deafdd498bc292e0d7)).

trgwii commented on server/utils/utils.js:
```diff
@@ -197,6 +204,10 @@ const preservedURLs = [
   "pricing"
 ];
 
+function getPath(path) {
+  return `${env.BASE_PATH}${path}`;
+}
+
```
It would be nicer if Kutt would validate BASE_PATH on startup and construct a URL object. getPath could then be implemented as something like `new URL(path, base_path.href)`

trgwii commented on README.md:
```diff
@@ -102,6 +102,8 @@ You can use files for each of the variables by appending `_FILE` to the name of
 | `SITE_NAME` |  Name of the website | `Kutt` | `Your Site` |
 | `DEFAULT_DOMAIN` |  The domain address that this app runs on | `localhost:3000` | `yoursite.com` |
+| `BASE_PATH` | The path the service will run on | `/` | `/url-shortener` |
+| `SHORT_URLS_INCLUDE_PATH` | Whether the shortened links will use the base path. If this value is false and BASE_PATH is specified, your proxy will need to route to the BASE_PATH of the service. | `false` | `true` |
```
Why is this also included, shouldn't this always be hardcoded to be true effectively?

trgwii commented on server/server.js:
```diff
@@ -69,24 +74,36 @@ app.set("views", [
+  router.get("*", renders.notFound);
+}
+// configure to run on the specified base path
+app.use(env.BASE_PATH, router);
```
Was this tested with BASE_PATH unset as well?
